docs/source/format/Intro.rst

```diff
@@ -0,0 +1,510 @@
+* Trading time can be represented as a Timestamp with metadata indicating the market
+  trading calendar
+
+Extension types can be defined by annotating any of the built-in Arrow data types
+(the “storage type”) with a custom type name and optional serialized representation
+(``'ARROW:extension:name'`` and ``'ARROW:extension:metadata'`` keys in the Field
+metadata structure).
+
+.. seealso::
+   The :ref:`format_metadata_extension_types` documentation.
+
+Canonical Extension Types
+-------------------------
+
+It is beneficial to share the definitions of well-known extension types so as to
+improve interoperability between different systems integrating Arrow columnar data.
+For this reason canonical extension types are defined in Arrow itself.
+
+Examples:
+
+* Fixed and variable shape tensor
+
+  - :ref:`fixed_shape_tensor_extension`
+  - :ref:`variable_shape_tensor_extension`
+
+.. seealso::
+   The :ref:`format_canonical_extensions` documentation.
+
+Community Extension Types
+-------------------------
+These are Arrow extension types that have been established as standards within specific domain areas.
+
+Example:
+
+* `GeoArrow`_: A collection of Arrow extension types for representing vector geometries
+
+.. _GeoArrow: https://github.com/geoarrow/geoarrow
+
+The Arrow C Data Interface
+==========================
+
+Arrow memory layout is meant to be a universal standard for tabular data, not tied to a specific
+implementation.
+
+While there are specifications to share Arrow data between processes or over the network (e.g. the
+IPC messages), the Arrow C Data Interface is meant to actually zero-copy share the data between
+different libraries within the same process (i.e. actually share the same buffers in memory).
+
+The Arrow C Data Interface defines a set of small C structures:
+
+.. code-block::
+
+   struct ArrowSchema {
+    const char* format;
+    const char* name;
+    const char* metadata;
+    int64_t flags;
+    int64_t n_children;
+    struct ArrowSchema** children;
+    struct ArrowSchema* dictionary;
+
+    // Release callback
+    void (*release)(struct ArrowSchema*);
+    // Opaque producer-specific data
+    void* private_data;
+   };
+
+   struct ArrowArray {
+    int64_t length;
+    int64_t null_count;
+    int64_t offset;
+    int64_t n_buffers;
+    int64_t n_children;
+    const void** buffers;
+    struct ArrowArray** children;
+    struct ArrowArray* dictionary;
+
+    // Release callback
+    void (*release)(struct ArrowArray*);
+    // Opaque producer-specific data
+    void* private_data;
+   };
+
+The C Data Interface passes Arrow data buffers through memory pointers. So, by construction, it allows
+you to share data from one runtime to another without copying it. Since the data is in standard Arrow
+in-memory format, its layout is well-defined and unambiguous.
+
+.. seealso::
+   The :ref:`c-data-interface` documentation.
+
+nanoarrow
```

Comment:
Similarly, I think mentioning nanoarrow was great for our live tutorial, but for this generic intro on the official docs of the full Arrow project, it seems a bit strange to call out one specific implementation?